File: src/main/java/pl/com/digita/kotlinexamples/wrapping/Person.java
package pl.com.digita.kotlinexamples.wrapping;

public class Person implements IPerson
{
    private String name;
    private String lastName;

    public Person()
    {
    }

    public Person(Person person)
    {
        this.name = person.name;
        this.lastName = person.lastName;
    }

    @Override
    public String getName()
    {
        return name;
    }

    @Override
    public void setName(String name)
    {
        this.name = name;
    }

    @Override
    public String getLastName()
    {
        return lastName;
    }

    @Override
    public void setLastName(String lastName)
    {
        this.lastName = lastName;
    }
}

File: src/main/java/pl/com/digita/kotlinexamples/wrapping/kotlin/Person.kt
package pl.com.digita.kotlinexamples.wrapping.kotlin

import java.math.BigDecimal

interface IPerson{
    var name: String
    var lastName: String
}

class Person(override var name: String, override var lastName: String) : IPerson

open class ForwardingPerson(private val person: IPerson): IPerson by person

class PersonWithAccountBallance(person: IPerson, var ballance: BigDecimal): ForwardingPerson(person)